cluster-service/deploy/templates/azure-operators-managed-identities-config.configmap.yaml

```diff
@@ -8,52 +8,102 @@ data:
         minOpenShiftVersion: 4.18
+        roleDefinitions:
+        {{- range .Values.azureOperatorsMI.controlPlane.roles }}
+          - resourceId: '/providers/Microsoft.Authorization/roleDefinitions/{{ .id }}'
+            name: '{{ .name }}'
+        {{- end }}
         azureRoleDefinitionResourceId: '/providers/Microsoft.Authorization/roleDefinitions/{{ .Values.azureOperatorsMI.controlPlane.roleId }}'
         azureRoleDefinitionName: '{{ .Values.azureOperatorsMI.controlPlane.roleName }}'
         optional: false
       cloud-controller-manager:
         minOpenShiftVersion: 4.18
+        roleDefinitions:
+        {{- range .Values.azureOperatorsMI.cloudControllerManager.roles }}
+          - resourceId: '/providers/Microsoft.Authorization/roleDefinitions/{{ .id }}'
+            name: '{{ .name }}'
+        {{- end }}
         azureRoleDefinitionResourceId: '/providers/Microsoft.Authorization/roleDefinitions/{{ .Values.azureOperatorsMI.cloudControllerManager.roleId }}'
         azureRoleDefinitionName: '{{ .Values.azureOperatorsMI.cloudControllerManager.roleName }}'
         optional: false
       ingress:
         minOpenShiftVersion: 4.18
+        roleDefinitions:
+        {{- range .Values.azureOperatorsMI.ingress.roles }}
+          - resourceId: '/providers/Microsoft.Authorization/roleDefinitions/{{ .id }}'
+            name: '{{ .name }}'
+        {{- end }}
         azureRoleDefinitionResourceId: '/providers/Microsoft.Authorization/roleDefinitions/{{ .Values.azureOperatorsMI.ingress.roleId }}'
         azureRoleDefinitionName: '{{ .Values.azureOperatorsMI.ingress.roleName }}'
         optional: false
       disk-csi-driver:
         minOpenShiftVersion: 4.18
+        roleDefinitions:
+        {{- range .Values.azureOperatorsMI.diskCsiDriver.roles }}
+          - resourceId: '/providers/Microsoft.Authorization/roleDefinitions/{{ .id }}'
+            name: '{{ .name }}'
+        {{- end }}
         azureRoleDefinitionResourceId: '/providers/Microsoft.Authorization/roleDefinitions/{{ .Values.azureOperatorsMI.diskCsiDriver.roleId }}'
         azureRoleDefinitionName: '{{ .Values.azureOperatorsMI.diskCsiDriver.roleName }}'
         optional: false
       file-csi-driver:
         minOpenShiftVersion: 4.18
+        roleDefinitions:
+        {{- range .Values.azureOperatorsMI.fileCsiDriver.roles }}
+          - resourceId: '/providers/Microsoft.Authorization/roleDefinitions/{{ .id }}'
+            name: '{{ .name }}'
+        {{- end }}
         azureRoleDefinitionResourceId: '/providers/Microsoft.Authorization/roleDefinitions/{{ .Values.azureOperatorsMI.fileCsiDriver.roleId }}'
         azureRoleDefinitionName: '{{ .Values.azureOperatorsMI.fileCsiDriver.roleName }}'
         optional: false
       image-registry:
         minOpenShiftVersion: 4.18
+        roleDefinitions:
+        {{- range .Values.azureOperatorsMI.imageRegistry.roles }}
+          - resourceId: '/providers/Microsoft.Authorization/roleDefinitions/{{ .id }}'
```

Comment:
Would the variable be .id or .resourceId?

Reply:
I just put `id` thinking of the context .. however, `.resourceId` should be good.

Reply:
I am sorry .. actually, this is roleId .. and since this is the array of roleIds and role names .. I chose just `id` and `name`. I am for now reverting back to `id` from `resourceId`